versioned/tiered-tests/src/main/java/com/dremio/nessie/versioned/tests/AbstractTestStore.java

```diff
@@ -268,18 +255,29 @@ public void putToCorrectBranchCollection() {
     putConditional(sample, ValueType.REF, true, Optional.of(condition));
   }
 
+  @Test
+  public void putFailingConditionExpression() {
+    final InternalRef sample = SampleEntities.createBranch(random);
+    final Entity nonExistent = Entity.ofString("Missing");
+    final InternalRef.Type type = InternalRef.Type.BRANCH;
+    final ConditionExpression condition = ConditionExpression.of(
+      ExpressionFunction.equals(ExpressionPath.builder(InternalRef.TYPE).build(), type.toEntity()),
+      ExpressionFunction.equals(ExpressionPath.builder("commit").build(), nonExistent));
+    putConditional(sample, ValueType.REF, false, Optional.of(condition));
+  }
+
   @Test
   public void putToIncorrectCollection() {
-    InternalRef sample = SampleEntities.createTag(random);
-    Id id = sample.getId();
-    InternalRef.Type type = InternalRef.Type.TAG;
+    final InternalRef sample = SampleEntities.createTag(random);
+    final Id id = sample.getId();
+    final InternalRef.Type type = InternalRef.Type.TAG;
     final ConditionExpression condition = ConditionExpression.of(
       ExpressionFunction.equals(
         ExpressionPath.builder(InternalRef.TYPE).build(),
         type.toEntity()),
       ExpressionFunction.equals(ExpressionPath.builder("commit").build(),
-        id.toEntity())
-    );
+        id.toEntity()));
+    // Trying to put a TAG entity to the VALUE collection should fail.
     assertThrows(IllegalArgumentException.class, () -> store.put(ValueType.VALUE, sample, Optional.of(condition)));
```

Comment:
Do you need to have a condition to test this case? It seems that this should also be more exhaustive, why only testing tag to value?

Reply:
This test morphed a bit. I'll refactor.